### ISSUE TITLE
Formatting of annotations may change semantic meaning

### ISSUE THREAD

**Issue by bethcutler** [facebookincubator/ktfmt#234]
Annotations are generally moved to the same line, but this may change the scope of the annotation.

Original code:
```kotlin
fun <T : Any?> testCasting(test: T, foo: String, bar: String): T {
  if (!foo.isEmpty()) {
    @Suppress("UNCHECKED_CAST")
    bar.isEmpty() || return (null as T)
  }
  return test
}
```

Formatted code:
```kotlin
fun <T : Any?> testCasting(test: T, foo: String, bar: String): T {
  if (!foo.isEmpty()) {
    @Suppress("UNCHECKED_CAST") bar.isEmpty() || return (null as T)
  }
  return test
}
```

After formatting, the Suppress annotation only applies on the first half of the expression, not the part of the expression where the cast actually occurs.

**Comment by cgrushko** [comment]
cc @strulovich

**Comment by strulovich** [comment]
I guess this is a problem in both ways, since converting case 2 into case 1 would also be a bug no? Does this mean we must preserve spacing on annotations? (At least for expressions)?

**Comment by davidtorosyan** [comment]
So be43fdb9942c1db7c67912a1301ee0c510979d25 fixed #247, which I think is a duplicate of this, so closing this out as well. Please reopen if I'm wrong!
